forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt

```diff
@@ -182,6 +195,26 @@ interface DeviceAuthenticator {
         return Calendar.getInstance().time
     }
 
+    /** Validate custom claims
+    * @param  customClaims: A map of custom claims to be validated
+    * @return Boolean value indicating whether the custom claims are valid or not
+     */
+    fun validateCustomClaims(customClaims: Map<String, Any>): Boolean {
+        val registeredKeys = listOf(
+            JWTClaimNames.SUBJECT,
+            JWTClaimNames.EXPIRATION_TIME,
+            JWTClaimNames.ISSUED_AT,
+            JWTClaimNames.NOT_BEFORE,
+            JWTClaimNames.ISSUER,
+            CHALLENGE,
+            PLATFORM,
+            ANDROID_VERSION
+        )
+        val customKeys = customClaims.keys.toList()
```

Comment:
i am not sure you need that toList(), because customClaims.keys is a collection.

return customClaims.keys.intersect(registeredKeys).isEmpty()